backends/jwt_remote.go

```diff
@@ -229,9 +260,31 @@ func (o *remoteJWTChecker) jwtRequest(host, uri, token string, dataMap map[strin
 		tlsStr = "https://"
 	}
 
-	fullURI := fmt.Sprintf("%s%s%s", tlsStr, o.host, uri)
+	host = o.host
+	if host == "" && o.options.parseToken {
+		iss, err := getIssForToken(o.options, token, o.options.skipUserExpiration)
+		if err == nil {
+			if o.isHostWhitelisted(iss) {
+				host = iss
+			} else {
+				errorString := fmt.Sprintf("host %s obtained from jwt is not whitelisted; rejecting", iss)
+				log.Error(errorString)
+				return false, errors.New(errorString)
+			}
+		} else {
+			log.Printf("jwt remote get iss error: %s", err)
+		}
+	}
+
+	if host == "" {
+		errorString := fmt.Sprintf("cannot determine host for the authorization request for token %s", token)
+		log.Error(errorString)
+		return false, errors.New(errorString)
+	}
```

Comment:
You're double checking `host == ""`, so this could probably be refactored to something like:
```
if host == "" {
  if o.options.parseToken {
    // checking iss is in whitelist logic
  } else {
    // log error and bail out because we have no valid host, be it missing single one or not whitelisted
  }
}
```

Reply:
Yeah, valid point, thanks. I've refactored this piece of code into separate functions - IMHO, it looks much cleaner now.

BTW, note that even in the original version of your code you don't use "host" parameter in jwtRequest(). Maybe I'll remove it?

Reply:
Could you point me to that part of the code to check?

Reply:
Well, in the origin "jwtRequest()" you use server host only in two places:
https://github.com/iegomez/mosquitto-go-auth/blob/master/backends/jwt_remote.go#L232
https://github.com/iegomez/mosquitto-go-auth/blob/master/backends/jwt_remote.go#L234

In both places you use a class member (`o.host`), not a function parameter (`host`). And you call this function always with the same class member as a parameter:
https://github.com/iegomez/mosquitto-go-auth/blob/master/backends/jwt_remote.go#L157
https://github.com/iegomez/mosquitto-go-auth/blob/master/backends/jwt_remote.go#L184
https://github.com/iegomez/mosquitto-go-auth/blob/master/backends/jwt_remote.go#L212

Reply:
Ah, I see, my bad. Sorry about that. 🤦 
You're absolutely right that I was passing the `host` params from `o.host`, but then using `o.host` directly in the `jwtRequest` call, so it should be safe to remove it altogether from the arguments.

Reply:
So I could I bother you with that last bit to keep it all in this same PR and then we'll be ready to merge?
Sorry again.

Reply:
Nah, thank you for bearing with my late replies and doing all the work in this PR.